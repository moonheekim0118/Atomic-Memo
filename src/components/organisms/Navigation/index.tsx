import React from 'react';
import MemoData from '../../../model/memoData';
import ListItem from '../../molecules/ListItem';
import styled from 'styled-components';

interface Props {
  open: boolean;
  datas: Array<MemoData>;
}

const Navigation = ({ open, datas }: Props) => {
  const List = datas.map((v, i) => (
    <ListItem
      key={v.id}
      title={v.title}
      time={v.time}
      kind={v.kind}
      path={`/${v.id}`}
    />
  ));
  return <StyledNavigation open={open}>{List}</StyledNavigation>;
};

const StyledNavigation = styled.nav<Props>`
  display: flex;
  flex-direction: column;
  position: fixed;
  height: 100%;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  width: 280px;
  background-color: #fff;

  border: 1px solid #e0e0d1;
  transition: transform 0.5s ease;
  transform: ${(props) => (props.open ? 'translateX(0)' : 'translateX(-100%)')};
  overflow-y: auto;

  scrollbar-width: none;
  -ms-overflow-style: none;

  ::-webkit-scrollbar {
    width: 0px;
    background: transparent;
  }
`;

export default Navigation;
