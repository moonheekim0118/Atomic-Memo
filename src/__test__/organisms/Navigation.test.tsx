import 'jsdom-global/register';
import React from 'react';
import Navigation from '../../components/organisms/Navigation';
import { MemoryRouter } from 'react-router-dom';
import { mount } from 'enzyme';

describe('<Navigation/>', () => {
  let container = null;
  let mockData = {
    id: 'test',
    title: 'test',
    main: 'test',
    kind: 'test',
    time: 'test',
  };
  it('renders correctly', () => {
    container = mount(
      <MemoryRouter>
        <Navigation datas={[mockData]} type="memos" open={true} />
      </MemoryRouter>
    );
  });

  it('matches snpashot', () => {
    expect(container.html()).toMatchSnapshot();
  });
});
