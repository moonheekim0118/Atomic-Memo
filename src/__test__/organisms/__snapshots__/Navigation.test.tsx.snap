// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navigation/> matches snpashot 1`] = `"<nav open=\\"\\" class=\\"Navigation__StyledNavigation-sc-1qkfpw4-0 djPKAE\\"><a class=\\"ListItem__StyledListItem-sc-1i7a48o-0 dUQKlR\\" href=\\"/test\\"><div class=\\"MemoTitle__StyledMemoTitle-e33wpr-0 cPfMRA\\"><span width=\\"auto\\" display=\\"inline\\" class=\\"Span__StyledSpan-sc-10y5m6u-0 dZpPqp small\\"></span><span width=\\"5.5em\\" display=\\"inline\\" class=\\"Span__StyledSpan-sc-10y5m6u-0 bvWVQT small\\">test</span></div><span width=\\"auto\\" display=\\"inline\\" class=\\"Span__StyledSpan-sc-10y5m6u-0 iExBFZ small\\">test</span></a></nav>"`;
